### PR TITLE
Implement document types

### PR DESCRIPTION
**Pull request by AoNoOokami**
Merge of US1812 branch (document types)
https://tree.taiga.io/project/rero21-reroils/us/1812?milestone=280004

### CODE REVIEW

File: projects/admin/src/app/record/brief-view/documents-brief-view/documents-brief-view.component.html [hunk 1]
```diff
@@ -15,41 +15,45 @@
...
-    <ul class="list-unstyled" *ngIf="provisionActivityPublications.length > 0">
-      <li *ngFor="let publication of provisionActivityPublications">{{ publication.value }}</li>
-    </ul>
-  </ng-container>
-</ng-container>
+<article *ngIf="record" class="card flex-row border-0">
+  <shared-thumbnail
+    class="col-sm-2"
+    [record]="record"
+    isSmall=true>
```
Comment (zannkukai) [inline projects/admin/src/app/record/brief-view/documents-brief-view/documents-brief-view.component.html:22]: as you mentioned yesterday, does`isSmall` should have prantheses ?

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          if (identifier.type === 'bf:Isbn') {
+            this.isbn = identifier.value;
+          }
+        }
+        if (this.isbn) {
+          this.getCoverUrl();
+        }
+      }
+      if (this.isSmall) {
+        this.figureStyle = 'thumb-brief';
```
Comment (jma) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.ts:100]: this should be done on ngOnChage as `isSmall` is an input property.

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html [hunk 3]
```diff
@@ -1,105 +1,99 @@
...
+        <ng-container *ngIf="record.metadata.extent">
+          <div id="doc-extent">{{ record.metadata.extent }}</div>
+        </ng-container>
+
+        <!-- EDITION STATEMENT -->
+        <ng-container *ngIf="record.metadata.editionStatement && record.metadata.editionStatement.length > 0">
+          <ul class="list-unstyled mb-0">
+            <li *ngFor="let edition of getStatement(record.metadata.editionStatement); let i = index"
+            [attr.id]="i | idAttribute:{prefix: 'doc-edition-statement'}">
+            {{ edition.value }}
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html:93]: COSMETIC : indentation

File: projects/public-search/src/app/document-brief/document-brief.component.html [hunk 4]
```diff
@@ -16,62 +16,56 @@
...
-            </a>
-            </ng-template>
-            {{ last ? '' : '; ' }}
-            {{ last && record.metadata.contribution.length > 3 ? '; …' : '' }}
-          </li>
-        </ul>
+  <shared-thumbnail
+    class="col-sm-2"
+    [record]="record"
+    isSmall="true">
```
Comment (zannkukai) [inline projects/public-search/src/app/document-brief/document-brief.component.html:22]: same here ?

File: projects/admin/src/app/record/brief-view/documents-brief-view/documents-brief-view.component.html (hunk 1 above)
Comment (zannkukai) [reply projects/admin/src/app/record/brief-view/documents-brief-view/documents-brief-view.component.html:22]: https://gitter.im/rero/reroils-PO?at=6011903e24cd6b60d8fe7f24

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      }
+    });
+  }
+
+  /**
+   * On init hook
+   * Set default cover image, or get cover image if exists.
+   */
+  ngOnInit() {
+    if (this.record && this.record.metadata) {
```
Comment (jma) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.ts:80]: this should be done on ngOnChage as `record` is an input property.

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.html [hunk 6]
```diff
@@ -0,0 +1,37 @@
+<!--
+ RERO ILS UI
+ Copyright (C) 2019 RERO
```
Comment (zannkukai) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.html:3]: just before @iGormilhit : 2019 --> 2021

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.scss [hunk 7]
```diff
@@ -0,0 +1,42 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2019 RERO
```
Comment (zannkukai) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.scss:3]: same

**Review by jma — CHANGES_REQUESTED** [review]

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.scss [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+.thumb-brief
+img {
```
Comment (zannkukai) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.scss:19]: I think it's more readable to have a single-line notation. With CSS, when you use a new line is often to use a compete other tag.
```thumb-brief img {```

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.scss [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  max-height: 122px;
+  max-width: 80px;
+}
+
+@media (max-width: 768px) {
+.thumb-brief
+  img {
```
Comment (zannkukai) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.scss:26]: same

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.scss [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    max-width: 48px;
+  }
+}
+
+.thumb-detail
+img {
```
Comment (zannkukai) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.scss:32]: same

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.scss [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+  max-height: 244px;
+  max-width: 160px;
+}
+
+@media (max-width: 768px) {
+.thumb-detail
+  img {
```
Comment (zannkukai) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.scss:39]: same

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.spec.ts [hunk 12]
```diff
@@ -0,0 +1,46 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2020 RERO
```
Comment (zannkukai) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.spec.ts:3]: 2021

File: projects/shared/src/lib/view/thumbnail/thumbnail.component.ts [hunk 13]
```diff
@@ -0,0 +1,104 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2020 RERO
```
Comment (zannkukai) [inline projects/shared/src/lib/view/thumbnail/thumbnail.component.ts:3]: 2021

**Review by jma — APPROVED** [review]

**Pull request merged**